localstack-core/localstack/services/stepfunctions/asl/static_analyser/usage_metrics_static_analyser.py

```diff
@@ -0,0 +1,56 @@
+import logging
+
+import localstack.services.stepfunctions.usage as UsageMetrics
+from localstack.services.stepfunctions.asl.antlr.runtime.ASLParser import ASLParser
+from localstack.services.stepfunctions.asl.component.common.query_language import (
+    QueryLanguageMode,
+)
+from localstack.services.stepfunctions.asl.static_analyser.static_analyser import StaticAnalyser
+
+LOG = logging.getLogger(__name__)
+
+
+class UsageMetricsStaticAnalyser(StaticAnalyser):
+    @staticmethod
+    def process(definition: str) -> "UsageMetricsStaticAnalyser":
+        analyser = UsageMetricsStaticAnalyser()
+        analyser.analyse(definition=definition)
+
+        try:
```

Comment:
Do we need the try/catch here? `has_jsonata` is just a getter and we can expect the counters to work (we're not using try/catch anywhere else)
if we want to be as defensive, then we should rather cover the `analyse` step.

Reply:
We didn't want the metrics collection to be the reason why SFN was failing. Will just surround the whole block in a try-catch rather

Reply:
OK wrapped the whole process in a try-except